Client/TabManagement/Legacy/LegacyTabManager.swift

```diff
@@ -234,22 +234,28 @@ class LegacyTabManager: NSObject, FeatureFlaggable, TabManager, TabEventHandler
 
     public static func makeWebViewConfig(isPrivate: Bool, prefs: Prefs?) -> WKWebViewConfiguration {
         let configuration = WKWebViewConfiguration()
-        configuration.processPool = WKProcessPool()
-        let blockPopups = prefs?.boolForKey(PrefsKeys.KeyBlockPopups) ?? true
-        configuration.preferences.javaScriptCanOpenWindowsAutomatically = !blockPopups
-        // We do this to go against the configuration of the <meta name="viewport">
-        // tag to behave the same way as Safari :-(
-        configuration.ignoresViewportScaleLimits = true
-        if isPrivate {
-            configuration.websiteDataStore = WKWebsiteDataStore.nonPersistent()
-        } else {
-            configuration.websiteDataStore = WKWebsiteDataStore.default()
+        /*
+         Ecosia: Ensure main thread to avoid EXC_BAD_ACCESS issue
+         All properties setup as part of `configuration`
+         should be accessed from MainThread only.
+        */
+        ensureMainThread {
+            configuration.processPool = WKProcessPool()
+            let blockPopups = prefs?.boolForKey(PrefsKeys.KeyBlockPopups) ?? true
+            configuration.preferences.javaScriptCanOpenWindowsAutomatically = !blockPopups
+            // We do this to go against the configuration of the <meta name="viewport">
+            // tag to behave the same way as Safari :-(
+            configuration.ignoresViewportScaleLimits = true
+                if isPrivate {
+                    configuration.websiteDataStore = WKWebsiteDataStore.nonPersistent()
+                } else {
+                    configuration.websiteDataStore = WKWebsiteDataStore.default()
+                }
+            configuration.setURLSchemeHandler(InternalSchemeHandler(), forURLScheme: InternalURL.scheme)
+            //Ecosia: inject cookie when config is created to make sure they are present
+            let cookie = isPrivate ? Cookie.makeIncognito() : Cookie.makeStandard()
+            configuration.websiteDataStore.httpCookieStore.setCookie(cookie)
         }
-
-        configuration.setURLSchemeHandler(InternalSchemeHandler(), forURLScheme: InternalURL.scheme)
-        //Ecosia: inject cookie when config is created to make sure they are present
-        let cookie = isPrivate ? Cookie.makeIncognito() : Cookie.makeStandard()
-        configuration.websiteDataStore.httpCookieStore.setCookie(cookie)
         return configuration
```

Comment:
Does it make sense to return the configuration after the main block inside `ensureMainThread` is executed? Wouldn't we need to make the whole method async?

Reply:
@lucaschifino thanks for the question!
Let me provide you with more context.
The properties setup of the `WKWebViewConfiguration` needs to happen on the main thread only.
I would argue that throughout our codebase, most of the times this happens without enforcing any thread. This is an issue happening on v122 (maybe v122+) Firefox up until they removed the `LegacyTabManager`.
I had a second 💭 and re-tested the app with the new skipped tests as well and the issue is no longer in place. Hence reverting the code as it was so perhaps we'll have less issues to solve in the future.